midudev/codi.link
# feat(Root): add auto complete for html tags

## Description 
As a user I should be able to auto complete the html ending tag.



## List of changes 


* register a monaco registerCompletionItemProvider for the html language
* add a completion provider to auto complete html tags

## Steps to Test or Reproduce 

1. write a valid opening html tag, you should be able to see the popup when writing the '>' char.
2. press the tab key, the suggestion is applied in the html editor and the caret position should be placed in the position next to the '>' of the opening tag.




## Screenshots
### Before
![Screenshot from 2021-10-02 19-41-30](https://user-images.githubusercontent.com/6330366/135733552-de390c1e-9474-4e4d-a07a-2cb49b30d08d.png)
### After
![Screenshot from 2021-10-02 19-41-06](https://user-images.githubusercontent.com/6330366/135733546-540fbae2-8890-46ca-8267-bc23ba4e57d9.png)
![Screenshot from 2021-10-02 19-41-12](https://user-images.githubusercontent.com/6330366/135733548-f0f61a62-da00-416c-be32-1dc04419f3c1.png)

## Impacted Areas in Application 
HTML Editor
